src/anonymous-credentials/presentation.ts

```diff
@@ -113,15 +113,34 @@ export class Presentation extends Versioned {
 
       const revealedEncoded = Presentation.encodeRevealed(i, presentedCred, presentedCredSchema, flattenedSchema[0]);
 
-      if (maxAttribs < numAttribs) {
-        sigParams = sigParams.adapt(numAttribs);
-        maxAttribs = numAttribs;
+      const paramsClass = paramsClassByPublicKey(publicKeys[i]);
```

Comment:
This logic is duplicated in` presentation-builder.ts` as well.

Reply:
Well, this 224-line `verify` function has a lot of common code with the 437-line function `finalize` from the `presentation-builder`. I guess it might be refactored in a separate task as this doesn't affect anything.

Reply:
I mentioned this because it looked simple to abstract in a function that takes `sigParamsByScheme` and `paramsClass`

Reply:
I'll move this logic to a single function as it will be changed anyway